// @aws-sdk/client-sts/dist-cjs/endpoint/ruleset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ruleSet = void 0;
exports.ruleSet = {
    version: "1.0",
    parameters: {
        Region: {
            builtIn: "AWS::Region",
            required: false,
            documentation: "The AWS region used to dispatch the request.",
            type: "String",
        },
        UseDualStack: {
            builtIn: "AWS::UseDualStack",
            required: true,
            default: false,
            documentation: "When true, use the dual-stack endpoint. If the configured endpoint does not support dual-stack, dispatching the request MAY return an error.",
            type: "Boolean",
        },
        UseFIPS: {
            builtIn: "AWS::UseFIPS",
            required: true,
            default: false,
            documentation: "When true, send this request to the FIPS-compliant regional endpoint. If the configured endpoint does not have a FIPS compliant endpoint, dispatching the request will return an error.",
            type: "Boolean",
        },
        Endpoint: {
            builtIn: "SDK::Endpoint",
            required: false,
            documentation: "Override the endpoint used to send this request",
            type: "String",
        },
        UseGlobalEndpoint: {
            builtIn: "AWS::STS::UseGlobalEndpoint",
            required: true,
            default: false,
            documentation: "Whether the global endpoint should be used, rather then the regional endpoint for us-east-1.",
            type: "Boolean",
        },
    },
    rules: [
        {
            conditions: [
                {
                    fn: "aws.partition",
                    argv: [
                        {
                            ref: "Region",
                        },
                    ],
                    assign: "PartitionResult",
                },
            ],
            type: "tree",
            rules: [
                {
                    conditions: [
                        {
                            fn: "booleanEquals",
                            argv: [
                                {
                                    ref: "UseGlobalEndpoint",
                                },
                                true,
                            ],
                        },
                        {
                            fn: "booleanEquals",
                            argv: [
                                {
                                    ref: "UseFIPS",
                                },
                                false,
                            ],
                        },
                        {
                            fn: "booleanEquals",
                            argv: [
                                {
                                    ref: "UseDualStack",
                                },
                                false,
                            ],
                        },
                        {
                            fn: "not",
                            argv: [
                                {
                                    fn: "isSet",
                                    argv: [
                                        {
                                            ref: "Endpoint",
                                        },
                                    ],
                                },
                            ],
                        },
                    ],
                    type: "tree",
                    rules: [
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "ap-northeast-1",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "ap-south-1",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "ap-southeast-1",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "ap-southeast-2",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
      /ú	Š-  Ä-ú—-  Ä-ú
Ä/ú©  (RbÄà®        I`şÿÿÿDa ! R" “”d         
   •c     KĞ “   $S4–`$   L`   —$QgÒJz   shouldLabelRenderBelow   QfÂIÏ6   emoji-label-bottom  I’Dg            -  Ä-ú—Š©(RbÄà®        I`şÿÿÿDağ" l# “”b         •c     KĞ ”   $S<–`.   L`   —	ÕDi            -  Ä-úÄÃw9ù/ú©   (RbÄà®        I`şÿÿÿDa # æ# h˜c            Pc     KĞ •   $S,¨`   ]”De            Äbú ©  (RbÄà®        I`şÿÿÿDa$ *$ •–b          —c     KĞ –   ¤a      QdŞú+F   positioning G$S(`
   L`   E”Dd             -  © (RbÄà®        I`şÿÿÿDa¼$ % •–b          —c     KĞ —   1
L`   TQs>BÈ¯H   <div
                    class="reaction-menu-message"
                >QcÂyÄò   </div>  å$S,`   L`   ÅÕDe            -  Ä]ú©(RbÄà®        I`şÿÿÿDa¤% Ş% h˜b         Pc     KĞ ˜   “L`   8Qlâ/+   <msn-social-most-reactions :reactionCounts= $QgZ¡İ    :fluentEmojisEnabled=  $Qgò­ë¨    :maxNumberOfEmojis="   QbÒùv   "/> 	$S(¨`
   L`   0Qj~¯{1#   reactionCountsForMostReactionEmojis –Dd             -  © (RbÄà®        I`şÿÿÿDa~& Î& —b          ‘c     KĞ ™   $S(“`
   L`   =–Dd             -  © (RbÄà®        I`şÿÿÿDa ' 0' —b          ‘c     KĞ š   $S(“`
   L`   ,QiÖH²À   maxNumberOfEmojisMostReactions  –Dd             -  © (RbÄà®        I`şÿÿÿDa`' ¦' —b          ‘c     KĞ ›   1
(L`   4QkvöùÉ&   <a
        tab-index="0"
        href=  ´Q°ªÍ2#¦   
        target="_blank"
        part="reactions-count-button"
        class="icon-button reactions-count-button accessibility-outline"
        title=""
        role=  $QgšSl7   
        aria-label="    Qf¢ÓN;   "
        data-t="  QcN
A   "
    > TQs¦»ö&F   
    <span aria-hidden="true" part="reaction-count" class="icon-label   © Qfb®9İ   </span>
    </a>
   e$S(“`
   L`   •ÕDd             -  © (RbÄà®        I`şÿÿÿDa( 6( h˜b          Pc     KĞ œ   $S0¨`   L`   Qb–	|l   linkQböÿ”   img ‘Df             -  —Š© (RbÄà®        I`şÿÿÿDaˆ) È) ’“b          ”c     KĞ    $S8•`&   L`   a‘‘Dh            Ä-  Ã-Âcúùø©   (RbÄà®        I`şÿÿÿDaø) T* ’“b       @ ”c     KĞ    $S(•`
   L`   ,Qiâ¥X”    reactionsCountButtonTelemetryTag‘Dd             -  © (RbÄà®        I`şÿÿÿDa~* È* ’“b          ”c     KĞ Ÿ   $S4•`$   L`   …—‘Dg            -  U˜
Ä-kú©(RbÄà®        I`şÿÿÿDaî* B+ ’“b       ”c     KĞ     $S<•`2   L`   —¹IÕDi            -  —
Ä-kú˜Š© (RbÄà®        I`şÿÿÿDaŞ+ L, h˜b       Pc     KĞ ¡   $S(¨`
   L`   Ñ’Dd             -  © (RbÄà®        I`şÿÿÿDaV, , “”b          •c     KĞ ¢   1
4L`   Qe2n[ó   <button
        QŒ†YC—„   
        part="reactions-count-button"
        class="icon-button reactions-count-button accessibility-outline"
        @mouseenter= Qf’X   
        @keypress= Qeê~˜D   
        @click=LQq¦BX?   
        title=""
        role="button"
        aria-expanded=" $QgNr·2   "
        aria-label="  uQd:nw   "
    >
    Qc…£M   
       QeînÃË   
    </button>
 !$SP–`V   L`
   e
Éi
Ñ…ÕDn            -  —-Ä-  Â-øÂ^úø-Ä-
Â^úø©   (RbÄà®        I`şÿÿÿDa@. / h˜d       P P    Pc     KĞ £   $S0¨`   L`   å‘Df            -  ÄÂ^úø© (RbÄà®        I`şÿÿÿDaH/ z/ ’“b         ”c     KĞ ¤   $S@•`8   L`   Qc#œ   Enter   e
Qbï   key á‘Dj             Ä- Ã-ùlú™-Ä]ú©  (RbÄà®        I`şÿÿÿDa¦/ 0 ’“c	       €   ”c     KĞ ¥   $S0•`   L`   ÕÕDf            -  ÄÂ^úø© (RbÄà®        I`şÿÿÿDaD0 „0 h˜b         Pc     KĞ ¦   $S(¨`
   L`   …’Dd             -  © (RbÄà®        I`şÿÿÿDa1 81 “”b          •c     KĞ §   $S`–`x    L`   	$Qg¦öZ   AriaLabelSelectReaction ‰a‘$Qg¦3½û   AriaLabelUseArrowKeys   ÕDr             - Ã-ùwÄ9ú ÄÃ-Â-Ácùø÷	w9ú Ä9ú Ä- Ã-ùw9ú ©  (RbÄà®        I`şÿÿÿDaj1 L2 h˜d
       °€
  
    Pc     KĞ ¨   $S(¨`
   L`   ”Dd             -  © (RbÄà®        I`şÿÿÿDav2 ¶2 •–b          —c     KĞ ©   $SH`F   L`   …“”Dl            -  —-Ä
nú–-  U™-Änú©   (RbÄà®        I`şÿÿÿDaæ2 †3 •–b       #—c     KĞ ª   $S@`8   L`   ’“”Dj            -  —-Ä
nú–-Änú©  (RbÄà®        I`şÿÿÿDa²3 44 •–b       #—c     KĞ «   1
L`   <Qmî^/   <span part="reaction-count" class="icon-label"> $S(`
   L`   ÑÕDd             -  © (RbÄà®        I`şÿÿÿDa¢4 Ü4 h˜b          Pc     KĞ ¬   “PL`$   ,QiºxË×    <a
        tabindex="0"
        
lQyöƒ¸ª`   
        target="_blank"
        class="icon-button comments-count-button accessibility-outline  â âDQoáú5   "
        part="comments-count-button"
        title=   …q Qf"ò2½   "
        @click=    QfÊô¡,   
        @keydown=  
‰,Qir“­~    "
    ><div class="icon-wrapper ©HQpØ³;   </div>
        <span aria-hidden="true" class="icon-label"> (QhöÍ   </span>
    </a> 
    
     õ1 QfRš/   commentsCountButton $S8¨`&   L`   Qe¾y‚¦   isCommentsInPage©ÕDh            -  —Š-Ä]ú©   (RbÄà®        I`şÿÿÿDaÄ5 6 h˜b       @ Pc     KĞ ­   $S0”`   L`    Qf¦0ßõ   topCommentVisible    Qf–<   top-comment-visible I–Df             -  —Š© (RbÄà®        I`şÿÿÿDaä6 B7 —b          ‘c     KĞ ®   $S0”`   L`   m™I–Df             -  —Š© (RbÄà®        I`şÿÿÿDaJ7 ¶7 —b          ‘c     KĞ ¯   $S8”`&   L`   ’­Qdº°Æ   no-backplateI–Dh             -  —-—Š©   (RbÄà®        I`şÿÿÿDa¾7 L8 —b         ‘c     KĞ °   $S8¨`(   L`   ea QfZ05   commentTooltipTitle IÕDh            -  —
-Ä-úŠ©  (RbÄà®        I`şÿÿÿDa¼8 :9 h˜b       P Pc     KĞ ±   $S8•`&   L`   •–¥©Dh             -  —-–Š©   (RbÄà®        I`şÿÿÿDa\9 Æ9 ‘’b         “c     KĞ ²   $S\¨`p   (L`   $QgIÇø   noTrendingCommentEnabled$QgšG…Ô   getNoTopCommentString   a$Qg:ß<!   totalCommentsCountLabel 	5IÕDq(            -  —
-Â]øŠ(-Ä-Ã
Á-÷
Á-÷Â-ú–À_ø÷öù©  (RbÄà®        I`şÿÿÿDaö9 D; h˜d       @ P @ Pc     KĞ ³   $S4¨`    L`   $Qg*o‹ú   handleClickOnCommentIcone
”Dg            -  Ä-Â^úø©  (RbÄà®        I`şÿÿÿDal; À; •–b       @ —c     KĞ ´   $SH`L    L`   }‘eQc"à³÷   $emit   $Qg¶   on-comments-count-click ÕDl             Ä- Ã-ùkú™-—-ÄÂ^úø	©(RbÄà®        I`şÿÿÿDaê; š< h˜c          Pc     KĞ µ   $SH¨`J   L`   e
QeJõ»×   renderTopCommenti
 Qf\Ã   handleHoverOnComment•Dl            -  —-Ä-  Â-øÂ^úø-Ä]ú
© (RbÄà®        I`şÿÿÿDaÊ< r= –—c       P @ c     KĞ ¶   $S0‘`   L`   I•Df            -  ÄÂ^úø© (RbÄà®        I`şÿÿÿDa¢= ä= –—b         c     KĞ ·   $S4‘`"   L`   Qeúâ
   hasTopComment   (Qhú²“   topCommentIconTelemetryTag  (Qh^ö¼İ   commentsButtonTelemetryTag  •Dg             -  —-Š-© (RbÄà®        I`şÿÿÿDa> ¤> –—b       P c     KĞ ¸   $S0‘`   L`   u$Qg®.Ì   has-top-comment-marker  IÕDf             -  —Š© (RbÄà®        I`şÿÿÿDaê> F? h˜b          Pc     KĞ ¹   $S$¨`   ]–Dc             ©(RbÄà®        I`şÿÿÿDab? l? —ñ‘c     KĞ º   $S0’`   L`   QezÏzˆ   getCommentButton–Df            -  Ä^ú©   (RbÄà®        I`şÿÿÿDar? ¢? —b         ‘c     KĞ »   $S(’`
   L`   ù–Dd             -  © (RbÄà®        I`şÿÿÿDa"@ Z@ —b          ‘c     KĞ ¼   $S`’`v   0L`   •}Ä1
 L`   4Qk>T®×'   <button
    id="topCommentAccessButton_                             authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "sa-east-1",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "us-east-1",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "us-east-2",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "us-west-1",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "us-west-2",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [],
                            endpoint: {
                                url: "https://sts.{Region}.{PartitionResult#dnsSuffix}",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "{Region}",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                    ],
                },
                {
                    conditions: [
                        {
                            fn: "isSet",
                            argv: [
                                {
                                    ref: "Endpoint",
                                },
                            ],
                        },
                        {
                            fn: "parseURL",
                            argv: [
                                {
                                    ref: "Endpoint",
                                },
                            ],
                            assign: "url",
                        },
                    ],
                    type: "tree",
                    rules: [
                        {
                            conditions: [
                                {
                                    fn: "booleanEquals",
                                    argv: [
                                        {
                                            ref: "UseFIPS",
                                        },
                                        true,
                                    ],
                                },
                            ],
                            error: "Invalid Configuration: FIPS and custom endpoint are not supported",
                            type: "error",
                        },
                        {
                            conditions: [],
                            type: "tree",
                            rules: [
                                {
                                    conditions: [
                                        {
                                            fn: "booleanEquals",
                                            argv: [
                                                {
                                                    ref: "UseDualStack",
                                                },
                                                true,
                                            ],
                                        },
                                    ],
                                    error: "Invalid Configuration: Dualstack and custom endpoint are not supported",
                                    type: "error",
                                },
                                {
                                    conditions: [],
                                    endpoint: {
                                        url: {
                                            ref: "Endpoint",
                                        },
                                        properties: {},
                                        headers: {},
                                    },
                                    type: "endpoint",
                                },
                            ],
                        },
                    ],
                },
                {
                    conditions: [
                        {
                            fn: "booleanEquals",
                            argv: [
                                {
                                    ref: "UseFIPS",
                                },
                                true,
                            ],
                        },
                        {
                            fn: "booleanEquals",
                            argv: [
                                {
                                    ref: "UseDualStack",
                                },
                                true,
                            ],
                        },
                    ],
                    type: "tree",
                    rules: [
                        {
                            conditions: [
                                {
                                    fn: "booleanEquals",
                                    argv: [
                                        true,
                                        {
                                            fn: "getAttr",
                                            argv: [
                                                {
                                                    ref: "PartitionResult",
                                                },
                                                "supportsFIPS",
                                            ],
                                        },
                                    ],
                                },
                                {
                                    fn: "booleanEquals",
                                    argv: [
                                        true,
                                        {
                                            fn: "getAttr",
                                            argv: [
                                                {
                                                    ref: "PartitionResult",
                                                },
                                                "supportsDualStack",
                                            ],
                                        },
                                    ],
                                },
                            ],
                            type: "tree",
                            rules: [
                                {
                                    conditions: [],
                                    endpoint: {
                                        url: "https://sts-fips.{Region}.{PartitionResult#dualStackDnsSuffix}",
                                        properties: {},
                                        headers: {},
                                    },
                                    type: "endpoint",
                                },
                            ],
                        },
                        {
                            conditions: [],
                            error: "FIPS and DualStack are enabled, but this partition does not support one or both",
                            type: "error",
                        },
                    ],
                },
                {
                    conditions: [
                        {
                            fn: "booleanEquals",
                            argv: [
                                {
                                    ref: "UseFIPS",
                                },
                                true,
                            ],
                        },
                    ],
                    type: "tree",
                    rules: [
                        {
                            conditions: [
                                {
                                    fn: "booleanEquals",
                                    argv: [
                                        true,
                                        {
                                            fn: "getAttr",
                                            argv: [
                                                {
                                                    ref: "PartitionResult",
                                                },
                                                "supportsFIPS",
                                            ],
                                        },
                                    ],
                                },
                            ],
                            type: "tree",
                            rules: [
                                {
                                    conditions: [],
                                    type: "tree",
                                    rules: [
                                        {
                                            conditions: [
                                                {
                                                    fn: "stringEquals",
                                                    argv: [
                                                        "aws-us-gov",
                                                        {
                                                            fn: "getAttr",
                                                            argv: [
                                                                {
                                                                    ref: "PartitionResult",
                                                                },
                                                                "name",
                                                            ],
                                                        },
                                                    ],
                                                },
                                            ],
                                            endpoint: {
                                                url: "https://sts.{Region}.{PartitionResult#dnsSuffix}",
                                                properties: {},
                                                headers: {},
                                            },
                                            type: "endpoint",
                                        },
                                        {
                                            conditions: [],
                                            endpoint: {
                                                url: "https://sts-fips.{Region}.{PartitionResult#dnsSuffix}",
                                                properties: {},
                                                headers: {},
                                            },
                                            type: "endpoint",
                                        },
                                    ],
                                },
                            ],
                        },
                        {
                            conditions: [],
                            error: "FIPS is enabled but this partition does not support FIPS",
                            type: "error",
                        },
                    ],
                },
                {
                    conditions: [
                        {
                            fn: "booleanEquals",
                            argv: [
                                {
                                    ref: "UseDualStack",
                                },
                                true,
                            ],
                        },
                    ],
                    type: "tree",
                    rules: [
                        {
                            conditions: [
                                {
                                    fn: "booleanEquals",
                                    argv: [
                                        true,
                                        {
                                            fn: "getAttr",
                                            argv: [
                                                {
                                                    ref: "PartitionResult",
                                                },
                                                "supportsDualStack",
                                            ],
                                        },
                                    ],
                                },
                            ],
                            type: "tree",
                            rules: [
                                {
                                    conditions: [],
                                    endpoint: {
                                        url: "https://sts.{Region}.{PartitionResult#dualStackDnsSuffix}",
                                        properties: {},
                                        headers: {},
                                    },
                                    type: "endpoint",
                                },
                            ],
                        },
                        {
                            conditions: [],
                            error: "DualStack is enabled but this partition does not support DualStack",
                            type: "error",
                        },
                    ],
                },
                {
                    conditions: [],
                    type: "tree",
                    rules: [
                        {
                            conditions: [
                                {
                                    fn: "stringEquals",
                                    argv: [
                                        {
                                            ref: "Region",
                                        },
                                        "aws-global",
                                    ],
                                },
                            ],
                            endpoint: {
                                url: "https://sts.amazonaws.com",
                                properties: {
                                    authSchemes: [
                                        {
                                            name: "sigv4",
                                            signingName: "sts",
                                            signingRegion: "us-east-1",
                                        },
                                    ],
                                },
                                headers: {},
                            },
                            type: "endpoint",
                        },
                        {
                            conditions: [],
                            endpoint: {
                                url: "https://sts.{Region}.{PartitionResult#dnsSuffix}",
                                properties: {},
                                headers: {},
                            },
                            type: "endpoint",
                        },
                    ],
                },
            ],
        },
    ],
};
